pkg/services/sqlstore/dashboard_version.go

```diff
@@ -58,3 +64,72 @@ func GetDashboardVersions(query *m.GetDashboardVersionsQuery) error {
 	}
 	return nil
 }
+
+func DeleteExpiredVersions(cmd *m.DeleteExpiredVersionsCommand) error {
+	return inTransaction(func(sess *DBSession) error {
+		var expiredCount int64 = 0
+		var versions []DashboardVersionExp
+
+		// Don't clean up if user set versions_to_keep to 2147483647 (MaxInt32)
+		if versionsToKeep := setting.DashboardVersionsToKeep; versionsToKeep < math.MaxInt32 {
+			// Get dashboard ids to clean up
+			affectedDashboardsQuery := fmt.Sprintf(`SELECT dashboard_id FROM dashboard_version
+				GROUP BY dashboard_id HAVING COUNT(dashboard_version.id)>%d`, versionsToKeep)
+
+			err := sess.Table("dashboard_version").
+				Select("dashboard_version.id, dashboard_version.version, dashboard_version.dashboard_id").
+				Where(fmt.Sprintf("dashboard_id IN (%s)", affectedDashboardsQuery)).
```

Comment:
Sending values to the database using `fmt.Sprintf` should be avoided since it opens up sql injection. 
In this case, its safe since the value passed in the query is an int but I think we should be consistent. 

Its preferred to send the query as a first argument and the pass all the values as params. 
ex https://github.com/grafana/grafana/blob/master/pkg/services/sqlstore/alert.go#L132

I also think we merge this query with the variable since it's not used anywhere else.